Testing that ws_id will be copied over to delineated watersheds.

phargogh opened the pull request:
It turned out that the conditional I thought was preventing `ws_id` from being copied over had no meaningful effect at all.  I've updated the tests to verify that if there's a `ws_id` field in the source vector, it'll be copied over into the target vector.

Fixes #249

phargogh committed:
Testing that ws_id will be copied over to delineated watersheds.

RE:#249

phargogh committed:
Installing all dependencies on one line to reduce chance of DLL conflict. RE:#249

emlys commented on .github/workflows/pythonpackage.yml:
```diff
@@ -59,8 +59,10 @@ jobs:
+            # Installing all of these in the same line will force conda to
+            # resolve all requirements at once, reducing the chance of DLL
+            # conflicts.
+            conda install --file requirements.txt gdal==${{ matrix.gdal }} setuptools build
```
oh interesting! I'd be curious to read about how conda resolves dependencies when done all at once vs. sequentially. do you happen to have a reference for that?

phargogh replied:
I don't have a documentation reference for it, but it does appear to be the case just from debugging this PR.  [Here's the latest build on `main`](https://pipelines.actions.githubusercontent.com/bGFNKdWfd0FvLn6TeVKOltVTPj0bb20eF1nGnpYGH1aTzpEgid/_apis/pipelines/1/runs/819/signedlogcontent/21?urlExpires=2022-03-24T23%3A21%3A07.3392228Z&urlSigningMethod=HMACV1&urlSignature=VLgds40U8QnPWloFlJDFKO%2B6wS%2FUKbkf9xcZsKGxX1Q%3D), where if you look at the logs you can see that the latest version of GDAL is being installed (3.4.2), but is then being downgraded to the build matrix-defined version of GDAL, 3.2.2.  With this all in one line, we _only_ install the one version of GDAL.

emlys reviewed (approved):
@phargogh it looks like the macos tests are still failing with the gdal dylib issue. The changes are unrelated to that, but maybe take out the comment in `pythonpackage.yml` if it's unclear that combining the conda commands is helpful? (keeping them all on one line is nice even if it doesn't fix the problem tho)

phargogh commented:
Thanks for taking a look @emlys !  I still think keeping them all in the same line is helpful, even if there's still this GDAL DLL issue. I'll look into the DLL issue a bit more and write back when I've found a fix.

phargogh committed:
Auto-updating conda to see if it'll grab the latest lib version.

phargogh committed:
Trying to install libpoppler from homebrew.

This is an attempt to circumvent the poppler import issues from gdal.
RE:#249

phargogh committed:
Adding fontconfig to the build matrix to see if that resolves issues. RE: #249 #252

phargogh committed:
Fontconfig needs to be in a list if it's in the matrix. RE:#249 #252

phargogh committed:
Retrying the build with a specific poppler version.  RE: #249 #252

phargogh committed:
Installing specific fontconfig on mac.

I stumbled across this, which suggested this workaround:
https://github.com/kb47/pdf-poppler/issues/2#issuecomment-1063521221

RE:#249 #252

phargogh committed:
Removing homebrew popppler install.  Not needed. RE:#249 #252

phargogh commented:
@emlys I think this one is ready for re-review as well now that the mac tests are passing again.

dcdenu4 commented on .github/workflows/pythonpackage.yml:
```diff
@@ -59,10 +59,19 @@ jobs:
+        run: |
+            # Specific fontconfig needed for binary build issues on mac.
+            # See natcap/pygeoprocessing#252
+            conda install fontconfig=2.13.1
```
Oofta!

phargogh replied:
Yep, not ideal at all, but until various packages are rebuilt on `conda-forge` it's a suitable workaround!

emlys replied:
Sorry this slipped through the cracks! Now that it's been a while, do we still need this?

phargogh replied:
To the best of my knowledge, we do not!  Removed in 8c7eaeb so let's see what happens in the tests.

phargogh committed:
Removing fontconfig hack. RE:#249.

phargogh committed:
Updating build matrix to adjust DLL import issues.

Matching what we got to work in
https://github.com/natcap/pygeoprocessing/pull/261.

RE:#249
